org.eclipse.scanning.server/src/org/eclipse/scanning/server/servlet/AcquireRequestHandler.java

```diff
@@ -63,12 +63,12 @@ public AcquireRequest process(AcquireRequest bean) throws EventException {
 
 			bean.setStatus(Status.COMPLETE);
 			bean.setMessage(null);
-		} catch (ScanningException | InterruptedException e) {
+		} catch (Exception e) {
 			e.printStackTrace();
 			bean.setStatus(Status.FAILED);
 			bean.setMessage(e.getMessage());
 			logger.error("Cannot acquire data for detector " + getBean().getDetectorName(), e);
-			throw new EventException(e);
+			throw (e instanceof EventException) ? (EventException)e : new EventException(e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace the usage of the "instanceof" operator by a catch block. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1193)